### ISSUE TITLE
Connection-worker mapping currently disallows concurrent tranactions

### ISSUE THREAD

**Issue by gudzpoz** [gudzpoz/r2dbc-jdbc#5]
Currently:

```
Reactive Connection 1 \
Reactive Connection 2 -|----> A single JdbcWorker ---> who owns a single java.sql.Connection
         ...          -/
Reactive Connection 2 /
```

Ideally:

```
Reactive Connection 1 \                            / Connection 1
Reactive Connection 2 -|----> A single JdbcWorker |- Connection 2
         ...          -/                          \-     ...
Reactive Connection N /                            \ Connection N
```

Passing Connection back and forth between JdbcWorker and JdbcConnection should do the job. `j2wait` and `j2shared` will then be unnecessary.

**Comment by gudzpoz** [comment]
Not going to pool it since r2dbc-pool provides the feature already.

**Comment by gudzpoz** [comment]
Probably resolved by a17972f0f7d9ed94ef0041a18e1fa4a7ef090fb4
